File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int MessageStart = logLine.IndexOf(' ');
        return logLine.Substring(MessageStart + 1).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int LevelStart = logLine.IndexOf('[');
        int LevelEnd = logLine.IndexOf(']');
        return logLine.Substring(LevelStart + 1, LevelEnd - LevelStart - 1).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    int start = line.find(':');
    int end = line.size();
    return line.substr(start + 2, end - start - 2);
}

std::string log_level(std::string line) {
    int start = 1;
    int end = line.find(']');
    return line.substr(start, end - start);
}

std::string reformat(std::string line) {
    return message(line) + " (" + log_level(line) + ")";
}
}  // namespace log_line
